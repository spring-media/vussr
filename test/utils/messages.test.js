const { logDevSuccess, logDevFail, getErrors } = require('../../src/utils/messages');
const logger = require('../../src/logger');

jest.mock('../../src/logger');

function setup() {
  const errorA = new Error('Error A');
  const errorB = new Error('Error B');
  const errors = [errorA, errorB];
  return { errorA, errorB, errors };
}

test('logs dev build success properly', () => {
  logDevSuccess(8080);
  expect(logger.info.mock.calls).toMatchSnapshot();
});

test('logs dev build fail properly', () => {
  const errors = ['foo error'];
  logDevFail({ errors });
  expect(logger.info.mock.calls).toMatchSnapshot();
  expect(logger.error.mock.calls).toMatchSnapshot();
});

test('reads errors from stats', () => {
  const { errors } = setup();
  const stats = { errors };
  expect(getErrors(stats)).toEqual(errors);
});

test('reads errors from stats', () => {
  const { errors } = setup();
  const compilation = { errors };
  const stats = { compilation };
  expect(getErrors(stats)).toEqual(errors);
});

test('reads errors from stats', () => {
  const { errors } = setup();
  const stats = { errors };
  const statsContainingStats = { stats };
  expect(getErrors(statsContainingStats)).toEqual(errors);
});

test('reads errors from stats', () => {
  const { errors } = setup();
  const compilation = { errors };
  const stats = { compilation };
  expect(getErrors([stats, stats])).toEqual([...errors, ...errors]);
});
