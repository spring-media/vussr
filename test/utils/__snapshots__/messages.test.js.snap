// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logs dev build fail properly 1`] = `
Array [
  Array [
    Object {
      "message": "
  [32mâœ” Server has been updated[39m

  [2m- App running at[22m http://localhost:8080
  [2m- Example article[22m http://localhost:8080/article/5b5996b98760980001ed8831
",
      "prefix": false,
    },
  ],
  Array [
    Object {
      "message": "
  [31mâœ– Compilation has failed[39m
",
      "prefix": false,
    },
  ],
]
`;

exports[`logs dev build fail properly 2`] = `
Array [
  Array [
    "foo error",
  ],
]
`;

exports[`logs dev build success properly 1`] = `
Array [
  Array [
    Object {
      "message": "
  [32mâœ” Server has been updated[39m

  [2m- App running at[22m http://localhost:8080
  [2m- Example article[22m http://localhost:8080/article/5b5996b98760980001ed8831
",
      "prefix": false,
    },
  ],
]
`;
